src/content/en/updates/2018/10/background-fetch.md

```diff
@@ -0,0 +1,448 @@
+      <td><code>Promise&lt;Response&gt;</code><br>The fetched response.
+      <p>The response is behind a promise because it may not have been received yet. The promise
+      will reject if the fetch fails.</p></td>
+    </tr>
+  </tbody>
+</table>
+
+### Service worker events
+
+<table class="responsive">
+  <tbody>
+    <tr><th colspan=2>Events</th></tr>
+    <tr>
+      <td><code>backgroundfetchsuccess</code></td>
+      <td>Everything was fetched successfully.</td>
+    </tr>
+    <tr>
+      <td><code>backgroundfetchfailure</code></td>
+      <td>One or more of the fetches failed.</td>
+    </tr>
+    <tr>
+      <td><code>backgroundfetchabort</code></td>
+      <td>One or more fetches failed.
+      <p>This is only really useful if you want to perform clean-up of related data.</p></td>
+    </tr>
+    <tr>
+      <td><code>backgroundfetchclick</code></td>
+      <td>The user clicked on the download notification.</td>
+    </tr>
+  </tbody>
+</table>
+
+The event objects have the following:
+
+<table class="responsive">
+  <tbody>
+    <tr><th colspan=2>Properties</th></tr>
+    <tr>
+      <td><code>registration</code></td>
+      <td><code>BackgroundFetchRegistration</code></td>
+    </tr>
+    <tr><th colspan=2>Methods</th></tr>
+    <tr>
+      <td><code>updateUI({ title, icons })</code></td>
+      <td>Lets you change the title/icons you initially set. This is optional, but it lets you
+      provide more context if necessary. You can only do this during
+      <code>backgroundfetchsuccess</code> and <code>backgroundfetchfailure</code> events.</td>
+    </tr>
+  </tbody>
+</table>
+
+### Reacting to success/failure
+
+We've already seen the `progress` event, but that's only useful while the user has a page open to
+your site. The main benefit of background fetch is things continue to work after the user leaves the
+page, or even closes the browser.
+
+If the background fetch successfully completes, your service worker will receive the
+`backgroundfetchsuccess` event, and `event.registration` will be the background fetch registration.
+
+After this event, the fetched requests and responses are no longer accessible, so if you want to
+keep them, move them somewhere like the [cache
+API](https://developer.mozilla.org/en-US/docs/Web/API/CacheStorage).
+
+As with most service worker events, use `event.waitUntil` so the service worker knows when the event
+is complete.
+
+For example, in your service worker:
+
+    addEventListener('backgroundfetchsuccess', (event) => {
+      const bgFetch = event.registration;
+
+      event.waitUntil(async function() {
+        // Create/open a cache.
+        const cache = await caches.open('downloads');
+        // Get all the records.
+        const records = await bgFetch.matchAll();
+        // Copy each request/response across.
+        const promises = records.map(async (record) => {
+          const response = await record.responseReady;
+          await cache.put(record.request, response);
```

Comment:
Would it make sense to (eventually) think of something like `Cache.putAll`? This is really expensive for large background fetches.